client-v2/src/shared/util/__tests__/shared.spec.ts

```diff
@@ -147,12 +148,36 @@ describe('Shared utilities', () => {
     });
 
     it('should create a single group with with all article fragments when the collection config has no groups', () => {
+      const result = normaliseCollectionWithNestedArticles(
+        collectionWithoutGroups,
+        {
+          ...collectionConfig,
+          groups: undefined
+        }
+      );
+      const groupId = result.normalisedCollection.live![0];
+      expect(result.groups[groupId].articleFragments).toHaveLength(3);
+    });
+    it('should create different groups for article fragments belonging to different groups even if they are not specificied in the config', () => {
       const result = normaliseCollectionWithNestedArticles(collection, {
         ...collectionConfig,
         groups: undefined
       });
-      const groupId = result.normalisedCollection.live![0];
-      expect(result.groups[groupId].articleFragments).toHaveLength(3);
+      const groupId1 = result.normalisedCollection.live![0];
+      expect(result.groups[groupId1].articleFragments).toHaveLength(1);
+      const groupId2 = result.normalisedCollection.live![1];
+      expect(result.groups[groupId2].articleFragments).toHaveLength(2);
+    });
+    it("should create empty groups for groups in the config which don't have collections in them", () => {
```

Comment:
Do you mean `collections` here, or collectionItems/articleFragments?

Reply:
articleFragments... Good catch